src/main/java/com/backend/webproject/dao/ProductDAO.java

```diff
@@ -9,18 +9,27 @@
 import com.backend.webproject.mappers.ProductMapper;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ProductDAO {
+
+    JdbcTemplate temp;
+
+    public ProductDAO(JdbcTemplate temp) {
+        this.temp = temp;
+    }
+
     @Autowired
     private NamedParameterJdbcTemplate jdbcTemplate;
+
```

Comment:
Add the Autowire annotation to the temp variable, erase the constructor and the jdbcTemplate variable and replace jdbcTemplate with temp.